// nba.js/lib/api/stats/utils/fetch.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = fetch;

var _got = require('got');

var _got2 = _interopRequireDefault(_got);

var _constants = require('./../constants');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

/**
 * Make a request to the stats API and return an error-first callback with the
 * JSON response.
 *
 * @param {string} endpoint - Optional API endpoint
 * @param {Object|Function} opts - Optional object of request options
 * @return {Promise} HTTP request response or error
 */
function fetch() {
  var endpoint = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : '';
  var opts = arguments[1];

  opts = Object.assign({
    headers: {
      host: 'stats.nba.com',
      connection: 'keep-alive',
      'user-agent': 'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_11_2) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/52.0.2743.116 Safari/537.36',
      referer: 'http://stats.nba.com/',
      Accept: 'text/html,application/xhtml+xml,application/xml;q=0.9,image/webp,*/*;q=0.8',
      'Accept-Encoding': 'gzip, deflate, sdch',
      'Accept-Language': 'en-US,en;q=0.8'
    },
    json: false
  }, opts || {});

  // eslint-disable-next-line no-useless-escape
  var re = /https?:\/\/(www\.)?[-a-zA-Z0-9@:%._\+~#=]{2,256}\.[a-z]{2,6}\b([-a-zA-Z0-9@:%_\+.~#?&//=]*)/;
  var url = re.test(endpoint) ? endpoint : '' + _constants.URL + endpoint;

  // in the case that endpoint is a URL, use endpoint, else concatenate the endpoint to URL
  return (0, _got2.default)(url, opts);
}
